Source/Particles/Deposition/ChargeDeposition.H

```diff
@@ -132,57 +132,57 @@ void doChargeDepositionShapeN (const GetParticlePosition& GetPosition,
             const amrex::Real x = (xp - xmin)*dxi;
 #endif
 
-            // Compute shape factor along x
-            // i: leftmost grid point that the particle touches
-            amrex::Real sx[depos_order + 1] = {0._rt};
             int i = 0;
-            if (rho_type[0] == NODE) {
-                i = compute_shape_factor(sx, x);
-            } else if (rho_type[0] == CELL) {
-                i = compute_shape_factor(sx, x - 0.5_rt);
+            if (depos_order > 1){
+                i = (depos_order%2==0) ? static_cast<int>(x+0.5_rt)-1_rt : static_cast<int>(x)-1_rt;
+            } else {
+                i = (depos_order%2==0) ? static_cast<int>(x+0.5_rt) : static_cast<int>(x);
```

Comment:
We probably do not need the suffix `_rt` for `1_rt` since the result of the operation is `int`. I also add a few white spaces to improve readability:
```suggestion
            if (depos_order > 1) {
                i = (depos_order%2 == 0) ? static_cast<int>(x+0.5_rt)-1 : static_cast<int>(x)-1;
            } else {
                i = (depos_order%2 == 0) ? static_cast<int>(x+0.5_rt) : static_cast<int>(x);
```
There will be similar suggestions throughout the code review.